### PR TITLE
add/port arrow tests

### PR DESCRIPTION
**Pull request by isao**
- autoload view-renderer.common
- ac deploy.server
- mw contextualizer & test (incl some add'l edge case handling)
- mw handler-tunnel
- mw router
- minor edits, chmod -x

### CODE REVIEW

**Comment by mridgway** [comment]
Why do you have to comment out stubbed modules? Other tests will break if you comment them out.

**Comment by isao** [comment]
actually I need to comment out more. omm the unit tests failed with the mojito-test.js on develop, and run without error when I comment out the modules with associated tests.

**Comment by isao** [comment]
works now without the commenting commit; backed it out

**Comment by isao** [comment]
@drewfish I backed out `//A.skip()` thanks

**Comment by isao** [comment]
the intermittent travis failure appears related to yuitest and the missing fixtures

File: lib/app/middleware/mojito-contextualizer.js [hunk 1]
```diff
@@ -106,15 +106,17 @@ RequestContextualizer.prototype = {
 
     _language: function(al, def) {
 
+        al = String(al || '').trim();
```
Comment (mojit0) [inline lib/app/middleware/mojito-contextualizer.js]: String wrappers and constructors like this are generally a "bad part" :). Probably best to just take "String" off and use:

(al || '').trim();

File: lib/app/middleware/mojito-contextualizer.js [hunk 2]
```diff
@@ -106,15 +106,17 @@ RequestContextualizer.prototype = {
...
+
         if (!al) {
             return def;
         }
 
-        var list = al.split(','),
+        var list = al.split(/ *, */), // accept-language value can have spaces
```
Comment (mojit0) [inline lib/app/middleware/mojito-contextualizer.js:115]: Probably need the regex(p) flag in a jslint directive if the test code is going to pass lint.

File: tests/base/mojito-test.js [hunk 3]
```diff
@@ -72,6 +72,18 @@ YUI.add('mojito-view-renderer', function(Y, NAME) {});
  * Add a mojito-test module containing the mocking support we want for other
  * Mojito components.
  */
+YUI.add('mojito-testutils', function(Y, NAME) {
+
+    /**
+     * @param {string} relpath relative to repo base directory
```
Comment (mojit0) [inline tests/base/mojito-test.js]: No description :)

File: tests/unit/lib/app/autoload/test-view-renderer.common.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+ * Copyright (c) 2011-2012, Yahoo! Inc.  All rights reserved.
+ * Copyrights licensed under the New BSD License.
+ * See the accompanying LICENSE file for terms.
+ */
+
+
+/*jslint anon:true, sloppy:true, nomen:true, node:true*/
+/*global YUI*/
+
+YUI().use('mojito-view-renderer', 'test', function(Y) {
```
Comment (mojit0) [inline tests/unit/lib/app/autoload/test-view-renderer.common.js:11]: I think arrow is suggesting we use Y.add() rather than use().

**Comment by mojit0** [comment]
A few nits, otherwise +1.

File: lib/app/middleware/mojito-contextualizer.js (hunk 1 above)
Comment (isao) [inline lib/app/middleware/mojito-contextualizer.js]: Originally I thought using the static function String() for type casting converted `undefined`, `NaN`, etc to `""`... It doesn't so I added the `||""`.. so yeah the function call doesn't really do anything except in the unlikely case that `al` is a number. I'll update.

File: lib/app/middleware/mojito-contextualizer.js (hunk 2 above)
Comment (isao) [inline lib/app/middleware/mojito-contextualizer.js:115]: jslint is fine with it (there's no dot or [^... )

File: tests/unit/lib/app/autoload/test-view-renderer.common.js (hunk 4 above)
Comment (isao) [inline tests/unit/lib/app/autoload/test-view-renderer.common.js:11]: ah geez. that change needs to be made to 40+ files and the README. leaving that another PR for now.

**Comment by mridgway** [comment]
@isao None of the middleware tests seem to run properly:

```
node /Users/mridgway/Projects/mojito/tests/../node_modules/yahoo-arrow/index.js /Users/mridgway/Projects/mojito/tests/unit/lib/app/middleware/*_descriptor.json --report=true --reportFolder=/Users/mridgway/Projects/mojito/tests/artifacts/arrowreport/unit/ --reuseSession --logLevel=WARN --browser=firefox

/Users/mridgway/Projects/mojito/node_modules/yahoo-arrow/node_modules/yui/dom-core/dom-core-min.js:7
",d="compareDocumentPosition",n=[],m=(function(){var q=e.config.doc.createElem
                                                                    ^
TypeError: Cannot call method 'createElement' of null
    at /Users/mridgway/Projects/mojito/node_modules/yahoo-arrow/node_modules/yui/dom-core/dom-core-min.js:7:262
    at Object.YUI.add.requires [as fn] (/Users/mridgway/Projects/mojito/node_modules/yahoo-arrow/node_modules/yui/dom-core/dom-core-min.js:7:382)
    at proto._attach (/Users/mridgway/Projects/mojito/node_modules/yahoo-arrow/node_modules/yui/yui-nodejs/yui-nodejs.js:749:37)
    at proto._use.handleLoader (/Users/mridgway/Projects/mojito/node_modules/yahoo-arrow/node_modules/yui/yui-nodejs/yui-nodejs.js:1065:33)
    at Object.YUI.add.Y.Loader._finish (/Users/mridgway/Projects/mojito/node_modules/yahoo-arrow/node_modules/yui/yui-nodejs/yui-nodejs.js:6790:19)
    at Object.YUI.add.Y.Loader._onSuccess (/Users/mridgway/Projects/mojito/node_modules/yahoo-arrow/node_modules/yui/yui-nodejs/yui-nodejs.js:6839:14)
    at Object.YUI.add.Y.Loader._insert.complete (/Users/mridgway/Projects/mojito/node_modules/yahoo-arrow/node_modules/yui/yui-nodejs/yui-nodejs.js:7036:24)
    at YUI.add.Y.Loader._insert.Y.Get.js.onSuccess (/Users/mridgway/Projects/mojito/node_modules/yahoo-arrow/node_modules/yui/yui-nodejs/yui-nodejs.js:7098:30)
    at YUI.add.pass (/Users/mridgway/Projects/mojito/node_modules/yahoo-arrow/node_modules/yui/yui-nodejs/yui-nodejs.js:3945:26)
    at YUI.add.Y.Get.js.check (/Users/mridgway/Projects/mojito/node_modules/yahoo-arrow/node_modules/yui/yui-nodejs/yui-nodejs.js:3969:21)
```

I get this error once for each test file.

**Comment by isao** [comment]
the middleware tests can't run in the browser
